lib/listener.js

```diff
@@ -110,7 +111,7 @@ internals.Listener.prototype._close = async function () {
 
     await Promise.all(Object.keys(this._sockets._items).map((id) => this._sockets._items[id].disconnect()));
 
-    this._wss.close();
+    await Util.promisify(this._wss.close);
```

Comment:
I think there are a couple things worth looking into here.  First, I don't believe `close()` is being called.  Instead it's awaiting a the function itself rather than a promise returned from said function.  Second, I think that the `close()` method will want to use `this` in its implementation, which isn't possible as written.  I think it's harder to explain than to just write some code that I would expect to see here.

```js
const close = Util.promisify(this._wss.close);
await close.call(this._wss);

Reply:
Aw, bummer! 🙏Good luck!  If I see anything I'll let you know.

Reply:
I think the issue is that the connected sockets are being told to close twice (either the server sends 2 close frames or the server sends one after already acknowledging the client is closing) and this then leads to a socket error and changes the whole cleanup process in nes to an errored cleanup rather than a clean close. This is definitely a timing issue, something is either faster or slower on HS. Adding small delays makes the problem go away all together but obviously isn't a proper solution.

Difficult to know if this is a bug in Ws or just our usage of it. I guess I need to go deeper into the websocket rabbit hole to find out.